spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/AbstractMessageListenerContainer.java

```diff
@@ -322,6 +332,23 @@ public void setDeBatchingEnabled(boolean deBatchingEnabled) {
 	}
 
 	/**
+	 * Public setter for the {@link Advice} to apply to listener executions. If {@link #setTxSize(int) txSize>1} then
+	 * multiple listener executions will all be wrapped in the same advice up to that limit.
+	 * <p>
+	 * If a {code #setTransactionManager(PlatformTransactionManager) transactionManager} is provided as well, then
+	 * separate advice is created for the transaction and applied first in the chain. In that case the advice chain
+	 * provided here should not contain a transaction interceptor (otherwise two transactions would be be applied).
+	 * @param adviceChain the advice chain to set
+	 */
+	public void setAdviceChain(Advice[] adviceChain) {
```

Comment:
Hm...
Someone has permissions to delete my comments.

OK. Will be serious.
You missed at least to answer me.
Thanks